block/manager.go

```diff
@@ -648,33 +654,23 @@ func (m *Manager) getCommit(header types.Header) (*types.Commit, error) {
 	}, nil
 }
 
-// IsProposer returns whether or not the manager is a proposer
-func (m *Manager) IsProposer() (bool, error) {
-	signerPubBytes, err := m.proposerKey.GetPublic().Raw()
-	if err != nil {
-		return false, err
-	}
-
-	return bytes.Equal(m.genesis.Validators[0].PubKey.Bytes(), signerPubBytes), nil
-}
-
 func (m *Manager) publishBlock(ctx context.Context) error {
 	select {
 	case <-ctx.Done():
 		return ctx.Err()
 	default:
 	}
 
-	isProposer, err := m.IsProposer()
-	if err != nil {
-		return fmt.Errorf("error while checking for proposer: %w", err)
-	}
-	if !isProposer {
+	if !m.isProposer {
+		m.logger.Debug("not a proposer, skipping block proposal")
```

Comment:
instead of `m.logger.Debug` and return nil, better to return it as `return errors.New("not a proposer")`

Reply:
Looks like the AggregationLoop is only run if the `aggregator` config is enabled. The caller of this also logs and continues. Updating.